common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,23 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
-      return text;
+    text = text.replace("''", "'");
```

Comment:
we only need to do this replacement if text is surrounded by single quotes. 
Similarly, if the text is surrounded by double quotes, we want to replace `""` with `"`.